tidy3d/components/medium.py

```diff
@@ -1085,3 +1094,171 @@ def plot(self, freqs: float, ax: Ax = None) -> Ax:
+            """Average `meds` with weights `weights`."""
+            eps_inf = 1
+            poles = []
+            for (med, weight) in zip(meds, weights):
+                eps_inf += weight * (med.eps_model(inf) - 1)
+                if isinstance(med, DispersiveMedium):
+                    poles += [(a, weight * c) for (a, c) in med.pole_residue.poles]
+                elif med.conductivity != 0:
+                    poles.append((0, weight * med.conductivity / (2 * EPSILON_0)))
+            return PoleResidue(eps_inf=np.real(eps_inf), poles=poles)
+
+        def get_background(comp: int) -> PoleResidue:
+            """Get the background medium appropriate for the `comp` component."""
+            meds = [get_component(neighbors[i], comp) for i in range(2)]
+            if comp == axis:
+                return meds[1]
+            weights = np.array(dls) / (np.sum(dls))
+            return weighted_avg(meds, weights)
+
+        dl = (dls[0] + dls[1]) / 2
+        media = [get_background(i) for i in range(3)]
+        media_fg = [self.ss, self.tt]
+        media_fg.insert(axis, Medium())
+        for i, med in enumerate(media_fg):
+            if i == axis:
+                continue
+            media[i] = weighted_avg([media[i], med], weights=[1, 1 / dl])
+        return AnisotropicMedium(xx=media[0], yy=media[1], zz=media[2])
+
+    @classmethod
+    def from_medium(cls, medium: Medium, dl: float) -> "Medium2D":
+        """Generate a ``Medium2D`` equivalent of a ``Medium`` of given thickness."""
+        sigma = medium.conductivity * dl
+        med = medium.updated_copy(conductivity=sigma)
+        return Medium2D(ss=med, tt=med)
+
+    @classmethod
+    def from_dispersive_medium(cls, medium: DispersiveMedium, dl: float) -> "Medium2D":
+        """Generate a ``Medium2D`` equivalent of a ``DispersiveMedium`` of a given thickness."""
+        med = medium.pole_residue
+        poles = [(p[0], p[1] * dl) for p in med.poles]
+        med = med.updated_copy(poles=poles)
+        return Medium2D(ss=med, tt=med)
+
+    @ensure_freq_in_range
+    def eps_model(self, frequency: float) -> complex:
+        """Complex-valued permittivity as a function of frequency."""
+        return np.mean(self.eps_diagonal(frequency=frequency))
+
+    @ensure_freq_in_range
+    def eps_diagonal(self, frequency: float) -> Tuple[complex, complex]:
+        """Main diagonal of the complex-valued permittivity tensor as a function of frequency."""
+
+        eps_ss = self.ss.eps_model(frequency)
+        eps_tt = self.tt.eps_model(frequency)
+        return (eps_ss, eps_tt)
+
+    @add_ax_if_none
+    def plot(self, freqs: float, ax: Ax = None) -> Ax:
+        """Plot n, k of a :class:`Medium` as a function of frequency."""
+
+        freqs = np.array(freqs)
+        freqs_thz = freqs / 1e12
+
+        for label, medium_component in zip(("ss", "tt"), (self.ss, self.tt)):
```

Comment:
could use that `.elements` property here as well if we add it. And in the `plot_sigma` method. (iterate through `label, medium_components in self.elements.items()`